packages/services/api/src/modules/audit-logs/providers/audit-logs-manager.ts

```diff
@@ -0,0 +1,310 @@
+      where.push(sql`timestamp >= ${from} AND timestamp <= ${to}`);
+    }
+    const whereClause = where.length > 0 ? sql`WHERE ${sql.join(where, ' AND ')}` : sql``;
+
+    const result = await this.clickHouse.query({
+      query: sql`
+        SELECT *
+        FROM audit_logs
+        ${whereClause}
+        ORDER BY timestamp DESC
+        LIMIT ${sqlFirst}
+        OFFSET ${sqlAfter}
+      `,
+      queryId: 'get-audit-logs',
+      timeout: 10000,
+    });
+
+    const data = AuditLogClickhouseArrayModel.parse(result.data);
+
+    return {
+      data,
+    };
+  }
+
+  async totalAuditLogs(organizationSlug: string): Promise<number> {
+    const isOwner = await this.auth.isOwnerOfOrganization({
+      organization: organizationSlug,
+    });
+    if (!isOwner) {
+      throw new GraphQLError('Unauthorized: You are not authorized to perform this action');
+    }
+
+    const result = await this.clickHouse.query({
+      query: sql`
+        SELECT count()
+        FROM audit_logs
+        WHERE organization_id = ${organizationSlug}
+      `,
+      queryId: 'total-audit-logs',
+      timeout: 10000,
+    });
+
+    const { count } = AuditLogTotalModel.parse(result.data[0]);
+
+    return count;
+  }
+
+  async exportAndSendEmail(
+    organizationSlug: string,
+    filter: { startDate?: Date | null; endDate?: Date | null },
+  ): Promise<{ ok: { url: string } | null } | { error?: { message: string } | null }> {
+    const isOwner = await this.auth.isOwnerOfOrganization({
+      organization: organizationSlug,
+    });
+    if (!isOwner) {
+      return {
+        error: {
+          message: 'Unauthorized: You are not authorized to perform this action',
+        },
+        ok: null,
+      };
+    }
+
+    const formattedStartDate = filter?.startDate ? new Date(filter.startDate) : undefined;
+    const formattedEndDate = filter?.endDate ? new Date(filter.endDate) : undefined;
+
+    try {
+      const { email } = await this.session.getViewer();
+      const totalAuditLogs = await this.totalAuditLogs(organizationSlug);
+      const getAllAuditLogs = await this.getPaginatedAuditLogs(
+        organizationSlug,
+        { first: totalAuditLogs ?? 1000, after: 0 },
+        { startDate: formattedStartDate, endDate: formattedEndDate },
+      );
+
+      if (!getAllAuditLogs || !getAllAuditLogs.data || getAllAuditLogs.data.length === 0) {
+        return {
+          ok: null,
+          error: {
+            message: 'No audit logs found for the given organization',
+          },
+        };
+      }
+
+      const data = getAllAuditLogs.data.map(log => {
+        const { id, timestamp, user_id, user_email, event_action, metadata } = log;
+        const newUser = {
+          id: metadata.user.id,
+          email: metadata.user.email,
+          provider: metadata.user.provider,
+          fullName: metadata.user.fullName,
+          displayName: metadata.user.displayName,
+        };
+        delete metadata.user;
```

Comment:
without that line, we leak data. Let's add structure to the `newMatadata` object. It can't be a combo of object spreads as we don't know what we spread